server/server.go

```diff
@@ -179,3 +177,98 @@ func (g *ReleaseManager) CheckForUpdate(p *Params) (res *Result, err error) {
 
 	return r, nil
 }
+
+type UpdateServer struct {
+	ReleaseManager *ReleaseManager
+
+	PublicAddr string
+	LocalAddr  string
+
+	PatchesDirectory string
+}
+
+func (u *UpdateServer) closeWithStatus(w http.ResponseWriter, status int) {
+	w.WriteHeader(status)
+	if status == http.StatusNoContent {
+		return
+	}
+	if _, err := w.Write([]byte(http.StatusText(status))); err != nil {
+		log.Debugf("Unable to write status %d: %v", status, err)
+	}
+}
+
+func (u *UpdateServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	var err error
+	var res *Result
+
+	if r.Method == "POST" {
+		defer r.Body.Close()
+
+		var params Params
+		decoder := json.NewDecoder(r.Body)
+
+		if err = decoder.Decode(&params); err != nil {
+			u.closeWithStatus(w, http.StatusBadRequest)
+			return
+		}
+
+		if res, err = u.ReleaseManager.CheckForUpdate(&params); err != nil {
+			if err == ErrNoUpdateAvailable {
+				u.closeWithStatus(w, http.StatusNoContent)
+				return
+			}
+			log.Debugf("CheckForUpdate failed. OS/Version: %s/%s, error: %q", params.OS, params.AppVersion, err)
+			u.closeWithStatus(w, http.StatusExpectationFailed)
+			return
+		}
+
+		log.Debugf("Got query from client %q/%q, resolved to upgrade to %q using %q strategy.", params.AppVersion, params.OS, res.Version, res.PatchType)
+
+		if res.PatchURL != "" {
+			res.PatchURL = u.PublicAddr + res.PatchURL
+		}
+
+		var content []byte
+		if content, err = json.Marshal(res); err != nil {
+			log.Debugf("Failed to marshal response: %s", err)
+			u.closeWithStatus(w, http.StatusInternalServerError)
+			return
+		}
+
+		nonce, _ := strconv.ParseInt(r.Header.Get("X-Message-Nonce"), 10, 64) // Can be zero for old clients.
+		hash := sha256.Sum256(append(content, []byte(fmt.Sprintf("%d", nonce))...))
```

Comment:
If the sole purpose of `nonce` is converting to `[]byte`, why bother converting to an integer at all? (If it's not cryptological significant).